sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -419,6 +419,10 @@ class Analyzer(override val catalogManager: CatalogManager) extends RuleExecutor
       ResolveRowLevelCommandAssignments ::
       MoveParameterizedQueriesDown ::
       BindParameters ::
+      ResolveEncodersInUDF ::
```

Comment:
I am moving the rules in to the main resolution batch, so that UDF encoders and table constraints can ben correctly handled in DML writes.
cc @viirya @Ngone51 @aokolnychyi

Reply:
Reverted this one.  It ends up failing many tests: https://github.com/gengliangwang/spark/runs/42517140406